### PR TITLE
Add PSK option

### PR DESCRIPTION
**Pull request by jkstrick**
Fix handshake segfault

### CODE REVIEW

File: src/DtlsSocket.cc [hunk 1]
```diff
@@ -128,6 +138,9 @@ DtlsSocket::DtlsSocket(const unsigned char *priv_key,
 	int ret;
 	const char *pers = "dtls_client";
 
+	recv_len = 0;
+	recv_buf = NULL;
```
Comment (pdulapalli) [inline src/DtlsSocket.cc:142]: Good catch

**Review by pdulapalli — DISMISSED** [review]

**Review by pdulapalli — DISMISSED** [review]
Based on behavior by consumers of this library occurring as expected, looks good to me

File: socket.js [hunk 2]
```diff
@@ -159,6 +159,8 @@ class DtlsSocket extends stream.Duplex {
 		if(!this.dontCloseDgramSocket) {
 			this.dgramSocket.close();
 		}
+
+		this.emit('close', this._hadError);
```
Comment (pdulapalli) [inline socket.js:163]: What's the upshot of making this change?

**Review by pdulapalli — DISMISSED** [review]

File: src/DtlsSocket.cc [hunk 3]
```diff
@@ -158,13 +173,18 @@ DtlsSocket::DtlsSocket(const unsigned char *priv_key,
 																		MBEDTLS_SSL_TRANSPORT_DATAGRAM,
 																		MBEDTLS_SSL_PRESET_DEFAULT);
 	if (ret != 0) goto exit;
-	mbedtls_ssl_conf_min_version(&conf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
+	mbedtls_ssl_conf_min_version(&conf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_2);
```
Comment (JSkubic-K4) [inline src/DtlsSocket.cc:176]: why do we use version 2 instead of 3?

Comment (jkstrick) [reply src/DtlsSocket.cc:176]: Same reason as the for node-mbed-dtls.  Z/IP Gateway requires support for TLS v1.2.

**Review by JSkubic-K4 — DISMISSED** [review]

**Review by pdulapalli — APPROVED** [review]

**Pull request merged**